jax/_src/numpy/lax_numpy.py

```diff
@@ -2586,7 +2600,7 @@ def _pad(array, pad_width, mode, constant_values, stat_length, end_values, refle
 (In numpy, "function" mode's argument should modify a rank 1 array in-place.)
 """)
 def pad(array, pad_width, mode="constant", **kwargs):
-  pad_width = _broadcast_to_pairs(pad_width, ndim(array), "pad_width")
+  pad_width = _broadcast_to_pairs(pad_width, _ndim_sequence(array), "pad_width")
```

Comment:
Isn't this the wrong fix? As a rule, don't we require numpy functions like `jnp.pad` to accept only arrays, not lists?

I think I would add a call to `_check_arraylike(array)` and fix any callers that are passing non-arrays here.

Reply:
Sounds good, I will fix the callers.